modules/apps/dynamic-data-mapping/dynamic-data-mapping-uad/src/main/java/com/liferay/dynamic/data/mapping/uad/display/DDMFormInstanceRecordUADDisplay.java

```diff
@@ -227,15 +228,62 @@ protected ThemeDisplay getThemeDisplay(
 			WebKeys.THEME_DISPLAY);
 	}
 
+	private int _getDDMFormInstanceRecordIndex(
+		DDMFormInstanceRecord ddmFormInstanceRecord) {
+
+		List<DDMFormInstanceRecord> ddmFormInstanceRecords =
+			_getDDMFormInstanceRecords(
+				ddmFormInstanceRecord.getFormInstanceId(),
+				ddmFormInstanceRecord.getUserId());
+
+		int index = 0;
+
+		for (DDMFormInstanceRecord currentDDMFormInstanceRecord :
+				ddmFormInstanceRecords) {
+
+			if (currentDDMFormInstanceRecord.getFormInstanceRecordId() ==
+					ddmFormInstanceRecord.getFormInstanceRecordId()) {
+
+				return index;
+			}
+
+			index++;
+		}
+
+		return -1;
+	}
+
+	private List<DDMFormInstanceRecord> _getDDMFormInstanceRecords(
+		long formInstanceId, long userId) {
+
+		DDMFormInstanceRecordUADUserCache ddmFormInstanceRecordUADUserCache =
+			null;
```

Comment:
@marcosapmf, what do you think to initialize `ddmFormInstanceRecordUADUserCache` with `_ddmFormInstanceRecordUADUserCacheMap.get(formInstanceId)`? I believe it can save some lines of code. 🙂

Reply:
@natocesarrego you are right! I'm going to do this right now :D